pkg/controllers/state/cluster.go

```diff
@@ -513,6 +513,14 @@ func (c *Cluster) populateInflight(ctx context.Context, n *StateNode) error {
 	if err != nil {
 		return client.IgnoreNotFound(err)
 	}
+	if n.NodeClaim != nil && n.Node != nil {
+		if initializedCond := n.NodeClaim.StatusConditions().GetCondition(v1beta1.Initialized); initializedCond.IsTrue() {
```

Comment:
If we're adding this change, we definitely need a test in the `state/suite_test.go` file to validate this behavior. Can you add the test case?